sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -21,149 +21,50 @@
 # SOFTWARE.
 
 
-
 """Http Logging Policy for Azure SDK"""
 
-import os
-import urllib
+import json
 import logging
-import types
-import ast
+import time
+from typing import Optional
 
+from azure.core.pipeline import PipelineRequest, PipelineResponse, HTTPRequestType, HTTPResponseType
 from azure.core.pipeline.policies import HttpLoggingPolicy
+from .http_constants import HttpHeaders
 
-
-
+def _format_error(payload: str) -> str:
+    output = json.loads(payload)
+    return output['message'].replace("\r", " ")
 class CosmosHttpLoggingPolicy(HttpLoggingPolicy):
 
-    def __init__(self, logger=None, **kwargs): # pylint: disable=unused-argument
-        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", None)
+    def __init__(self, logger: Optional[logging.Logger] = None, **kwargs): # pylint: disable=unused-argument
+        self._enable_diagnostics_logging = kwargs.pop("enable_diagnostics_logging", False)
         super().__init__(logger, **kwargs)
         if self._enable_diagnostics_logging:
-            self.logger = logger
+            cosmos_disallow_list = ["Authorization", "ProxyAuthorization"]
+            cosmos_allow_list = [
+                v for k, v in HttpHeaders.__dict__.items() if not k.startswith("_") and k not in cosmos_disallow_list
+            ]
+            self.allowed_header_names = set(cosmos_allow_list)
 
     def on_request(self, request): # pylint: disable=too-many-return-statements, too-many-statements
         # type: (PipelineRequest) -> None
-        http_request = request.http_request
         options = request.context.options
-        self._enable_diagnostics_logging = request.context.setdefault(
-            "enable_diagnostics_logging",
-            options.pop("enable_diagnostics_logging", self._enable_diagnostics_logging))
+        self._enable_diagnostics_logging = options.pop("enable_diagnostics_logging", self._enable_diagnostics_logging)
```

Comment:
We don't need this line, as this can't be passed on request.

Reply:
Without popping the option I receive an error saying that the request received an unexpected argument:
<img width="513" alt="image" src="https://user-images.githubusercontent.com/30335873/221003107-71e630e8-0688-4c2c-a72d-4e04c0b25ed1.png">
Happens with both sync and async

Reply:
This must mean that it's being passed in on the method call?

Reply:
I'm only passing it at the client constructor level:
<img width="390" alt="image" src="https://user-images.githubusercontent.com/30335873/221005238-028fc807-6ab4-47f9-bf1c-25595a2f08a7.png">